### ISSUE TITLE
helm-M-x broken

### ISSUE THREAD

**Issue by fm4d** [emacs-helm/helm#1415]
helm-M-x suddenly stopped working for me, when I select command in helm-M-x window nothing happens, window dissapears as usually but command is not run. There is nothing in messages or backtrace. Other helm commands and helmized commands seems to be fine, only helm-M-x and helm-run-external-command are behaving like this. Runing helm with `emacs-helm.sh` leads to same result. I updated helm from MELPA after 2-3 days so it must be something recent I guess.

**Comment by jkr** [comment]
The problem seems to have been introduced with d72d7dc (it works prior to that).

**Comment by gongzhitaao** [comment]
I couldn't even kill a buffer....

**Comment by elandx** [comment]
helm-M-x stopped working for me too, switched to an earlier version ...

**Comment by willtm** [comment]
I've encountered the same problem described by others:  helm-M-x no longer works.

**Comment by mordocai** [comment]
Looks like @xuchunyang fixed it in that pull request. Please merge oh maintainers!

**Comment by xuchunyang** [comment]
As a workaround, evaluate the following form:

```
(defun helm-execute-selection-action ()
  "Execute current action."
  (helm-log-run-hook 'helm-before-action-hook)
  ;; Position can be change when `helm-current-buffer'
  ;; is split, so jump to this position before executing action.
  (helm-current-position 'restore)
  (prog1 (helm-execute-selection-action-1)
    (helm-log-run-hook 'helm-after-action-hook)))
```

**Comment by thierryvolpiatto** [comment]
Chris Carpenter <notifications@github.com> writes:

> Looks like @xuchunyang fixed it in that pull request. Please merge oh maintainers!

The fix of @xuchunyang have now been merged, it fix #1412 #1414 and #1415.
Thanks to him!

-- 
Thierry
